### DISCUSSION TITLE
Enable SDR to HDR tone mapping via command line

### DISCUSSION THREAD

**Discussion by sorted01** [awawa-dev/HyperHDR#58] [Q&A] [answered]
Hey Awawa!
first of all thank you for your awesome work! 👍 
A short introtuction for my question:
I am using IoBroker to control my smart home. Within that I have the Linux control adapter which can send commands to the raspberry. 
Is it possible to enable & disable the SDR to HDR tone mapping by such an command?
My idea is to create a virtual button which can be triggered by a button/voice/whatever which sends a command line to my raspberry that changes the tone mapping.

Thank you!

***EDIT***

in the issues i've found this answer from you:

```
- platform: command_line
switches:
hyperion_toggle_hdr:
command_on: curl -X POST -i "http://192.168.178.92:8090/json-rpc" -d '{"command":"videomodehdr","HDR":1}'
command_off: curl -X POST -i "http://192.168.178.92:8090/json-rpc" -d '{"command":"videomodehdr","HDR":0}'
command_state: curl -k http://{IP_OF_HYPERHDR}:8090/json-rpc -H 'Content-Type: application/json' -d '{"command":"serverinfo"}'
value_template: >
{% if value_json.info.videomodehdr == 1 %}
{{ true }}
{% else %}
{{ false }}
{% endif %}
```

Am I right with that: To control it I only need "command_on" and "command_off"

Having the current state within IoBroker would be a nice to have. But therefore I guess I have limited knowledge of how to put the outcome of "command_state" into an IoBroker datapoint.

Best Regards
sorted

**Comment by awawa-dev** [comment, marked as answer]
Hi
Yes, you can control HDR settings with the commands that you have included.

API address:
http://{IP_OF_HYPERHDR}:8090/json-rpc

To set HDR tone mapping include JSON in the _videomodehdr_ request  (HDR: 0 - off, 1 -on). For example:
{"command":"videomodehdr","HDR":1}

To get the state of main components send:
{"command":"serverinfo"}

The result of _serverinfo_ command returns a json. Home Assistant  (https://www.home-assistant.io/docs/configuration/templating/) can automatically parse it to _value_json_ from the above example so to extract a HDR property from that object you can simple refer to _value_json.info.videomodehdr_ Unfortunately have no experience with ioBroker and could not find a parser for json but it offers regex parser that probably can extract that value in somehow unconventional way (ignore json structure and treat it as a text): https://github.com/ioBroker/ioBroker.parser

So for following result from the HyperHDR:
![obraz](https://user-images.githubusercontent.com/69086569/116098172-21b41300-a6ab-11eb-9eb1-9db7aa54c697.png)

you should use regex:
````"videomodehdr": (\d)````

to assign the state of HDR tone mapping to the final regex group (tested on https://regex101.com/):
![obraz](https://user-images.githubusercontent.com/69086569/116098935-c171a100-a6ab-11eb-8834-94f0f4e26a84.png)

**Reply by sorted01** [reply]
Hey, it's me again. Thank you very much for your feedback. Such an detailed answer! But... Before i get the output working I would need again some help with the control itself. :/ Sorry I am a real newb in using Linux...

As I did want to test it upfront I tried using the command line from above within PuTTY leading to following error:

> pi@HyperHDR:~ $ curl -X POST -i "http://192.168.178.92:8090/json-rpc" -d ''{"command":"videomodehdr","HDR":1}''
>
>HTTP/1.1 200 OK.
>Content-Type: application/json
>Date: Mon, 26 Apr 2021 19:51:16 UTC
>Content-Length: 138
>Server: HyperHDR Webserver
>
>{
>    "command": "",
>    "error": "Errors during message parsing, please consult the HyperHDR Log.",
>    "success": false,
>    "tan": 0
>}
>curl: (6) Could not resolve host: HDR 

This showed up within the HyperHDR Logs:
>2021-04-26T19:49:56.906Z [HTTPJSONRPC] (DEBUG) (JsonAPI.cpp:85:handleInstanceSwitch()) Client '::ffff:192.168.178.92' switch to HyperHDR instance 0
>2021-04-26T19:49:56.908Z [HTTPJSONRPC] (ERROR) Failed to parse json data from JsonRpc@::ffff:192.168.178.92: Error: illegal value at Line: 0, Column: 1
>2021-04-26T19:50:47.325Z [V4L2:/DEV/VIDEO0] (INFO) Video FPS: 29.78, av. delay: 6ms, good: 1787, bad: 0 (60.01,15)
>2021-04-26T19:51:01.606Z [HTTPJSONRPC] (DEBUG) (JsonAPI.cpp:85:handleInstanceSwitch()) Client '::ffff:192.168.178.92' switch to HyperHDR instance 0
>2021-04-26T19:51:01.606Z [HTTPJSONRPC] (ERROR) Failed to parse json data from JsonRpc@::ffff:192.168.178.92: Error: illegal value at Line: 0, Column: 1
2021-04-26T19:51:16.576Z [HTTPJSONRPC] (DEBUG) (JsonAPI.cpp:85:handleInstanceSwitch()) Client '::ffff:192.168.178.92' switch to HyperHDR instance 0
2021-04-26T19:51:16.577Z [HTTPJSONRPC] (ERROR) Failed to parse json data from JsonRpc@::ffff:192.168.178.92: Error: illegal value at Line: 0, Column: 1`

Again sorry for asking supid questions. But maybe you can support me once more ;)

Best Regards
sorteds

**Reply by awawa-dev** [reply]
No problem. I'm working on make the request easier but I'm not promising anything for the moment.
For your problem there are some unnecessary (doubled ' ) chars after "-d " and at the end of the command
Try this:

> curl -X POST -i "http://192.168.178.92:8090/json-rpc" -d '{"command":"videomodehdr","HDR":1}'

**Reply by sorted01** [reply]
Thank you so so much! It works like a charm! :)

EDIT:
I've updated the commands in the initial post. So that no one will copy that miskake
